yarn-project/simulator/src/public/side_effect_trace.ts

```diff
@@ -102,26 +102,20 @@ export class PublicSideEffectTrace implements PublicSideEffectTraceInterface {
     contractInstance: TracedContractInstance,
     contractClass: ContractClassIdPreimage,
   ) {
-    // Deduplicate - we might want a map here to make this more efficient
-    const idx = this.avmCircuitHints.contractBytecodeHints.items.findIndex(
-      hint => hint.contractInstanceHint.address === contractInstance.address,
+    const instance = new AvmContractInstanceHint(
```

Comment:
The de-duplication didn't really work here (but is a nice to have eventually), we'll re-introduce it once we have proper tracing across call boundaries